lib/assertions.ts

```diff
@@ -492,15 +492,41 @@ export default class DOMAssertions {
    * @see {@link #hasClass}
    */
   hasStyle(expected: object, message?: string): void {
+    this.hasPseudoElementStyle(null, expected, message);
+  }
+
+  /**
+   * @private
+   */
+  hasPseudoElementStyle(selector: null, expected: object, message?: string): void;
+
+  /**
+   * Assert that the pseudo element for `selector` of the [HTMLElement][] has the `expected` style declarations using
+   * [`window.getComputedStyle`](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle).
+   *
+   * @name hasPseudoElementStyle
+   * @param {string} selector
+   * @param {object} expected
+   * @param {string?} message
+   *
+   * @example
+   * assert.dom('.progress-bar').hasPseudoElementStyle(':after', {
+   *   content: '";"',
+   * });
+   *
+   * @see {@link #hasClass}
+   */
+  hasPseudoElementStyle(selector: string, expected: object, message?: string): void {
```

Comment:
I **think** this would have to be `selector: string | null`, and then we'd need another variant above that accepts `selector: string` as the public interface to that method. If I remember correctly the actual implementation is always hidden if you have specializations like the `selector: null` above 🤔

Reply:
Will do.